lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/resourcemanager/ResourceEventImpl.java

```diff
@@ -74,6 +78,11 @@ public String getRuntimeName() {
     return runtimeName;
   }
 
+  @Override
+  public Optional<Map<String, String>> getNodeLabelExpressions() {
+    return nodeLabelExpressions;
```

Comment:
This is potentially unsafe, as the caller of this API can subsequently change the returned `Optional` or `Map`.